{
  "active_lock_reason": null,
  "assignee": null,
  "assignees": [],
  "author_association": "CONTRIBUTOR",
  "body": "<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!\r\n\r\nIf the matter is security related, please disclose it privately via https://kubernetes.io/security/\r\n-->\r\n\r\n\r\n**What happened**:\r\n\r\nPods will not start on Windows nodes, even though nodes are ready and pods are scheduled to nodes.\r\n\r\nKubelet logs are filled with this:\r\n\r\n```\r\nE0306 18:20:36.813221    3564 runtime.go:78] Observed a panic: \"invalid memory address or nil pointer dereference\" (runtime error: invalid memory address or nil pointer dereference)\r\ngoroutine 206 [running]:\r\nk8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x38ac960, 0x6b90c30)\r\n\t/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xaa\r\nk8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)\r\n\t/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x89\r\npanic(0x38ac960, 0x6b90c30)\r\n\t/usr/local/go/src/runtime/panic.go:679 +0x1c0\r\nk8s.io/kubernetes/pkg/kubelet.(*Kubelet).canAdmitPod(0xc000640a80, 0x0, 0x0, 0x0, 0xc00102d400, 0x0, 0x0, 0x6a62da3, 0xa, 0x774)\r\n\t/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1794 +0x15d\r\nk8s.io/kubernetes/pkg/kubelet.(*Kubelet).HandlePodAdditions(0xc000640a80, 0xc000d5e1a8, 0x1, 0x1)\r\n\t/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:2071 +0x312\r\nk8s.io/kubernetes/pkg/kubelet.(*Kubelet).syncLoopIteration(0xc000640a80, 0xc000680ae0, 0x48cc5c0, 0xc000640a80, 0xc000816ea0, 0xc000816f00, 0xc0009f1800, 0xc0002f0601)\r\n\t/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1913 +0x561\r\nk8s.io/kubernetes/pkg/kubelet.(*Kubelet).syncLoop(0xc000640a80, 0xc000680ae0, 0x48cc5c0, 0xc000640a80)\r\n\t/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1856 +0x378\r\nk8s.io/kubernetes/pkg/kubelet.(*Kubelet).Run(0xc000640a80, 0xc000680ae0)\r\n\t/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1452 +0x26e\r\nk8s.io/kubernetes/cmd/kubelet/app.startKubelet.func1()\r\n\t/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubelet/app/server.go:1133 +0x43\r\nk8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000b352a0)\r\n\t/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x65\r\nk8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000b352a0, 0x4824e60, 0xc000b39cb0, 0xc0000bf601, 0xc0000800c0)\r\n\t/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xaa\r\nk8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000b352a0, 0x0, 0x0, 0x1, 0xc0000800c0)\r\n\t/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0xe9\r\nk8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc000b352a0, 0x0, 0xc0000800c0)\r\n\t/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x54\r\ncreated by k8s.io/kubernetes/cmd/kubelet/app.startKubelet\r\n```\r\n\r\n**What you expected to happen**:\r\n\r\n**How to reproduce it (as minimally and precisely as possible)**:\r\n\r\nBuild K8s from master, create any pod on Windows node.\r\n\r\n**Anything else we need to know?**:\r\n\r\nI strongly believe this commit is responsible: https://github.com/kubernetes/kubernetes/commit/ac32644d6e712af0747f39a945c8df29d6a9cfc4 \r\n\r\nWhat happened is that now, with the commit mentioned above, a nil podAdmitHandler is registered, that when called will result in a panic.\r\n\r\nMore specifically here: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L871\r\n\r\nFor Windows, `GetAllocateResourcesPodAdmitHandler` returns `nil`:\r\n\r\nhttps://github.com/kubernetes/kubernetes/blob/1afb554222605d4d67a911c9bbb0fa5b1049c2dc/pkg/kubelet/cm/container_manager_windows.go#L181\r\n\r\n\r\n\r\n**Environment**:\r\n\r\nAll K8s built from master. This affects all sig-windows master tests.",
  "closed_at": "2020-03-15T18:00:42Z",
  "closed_by": {
    "avatar_url": "https://avatars0.githubusercontent.com/u/20407524?v=4",
    "events_url": "https://api.github.com/users/k8s-ci-robot/events{/privacy}",
    "followers_url": "https://api.github.com/users/k8s-ci-robot/followers",
    "following_url": "https://api.github.com/users/k8s-ci-robot/following{/other_user}",
    "gists_url": "https://api.github.com/users/k8s-ci-robot/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/k8s-ci-robot",
    "id": 20407524,
    "login": "k8s-ci-robot",
    "node_id": "MDQ6VXNlcjIwNDA3NTI0",
    "organizations_url": "https://api.github.com/users/k8s-ci-robot/orgs",
    "received_events_url": "https://api.github.com/users/k8s-ci-robot/received_events",
    "repos_url": "https://api.github.com/users/k8s-ci-robot/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/k8s-ci-robot/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/k8s-ci-robot/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/k8s-ci-robot"
  },
  "comments": 11,
  "comments_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/88913/comments",
  "created_at": "2020-03-06T17:23:46Z",
  "events_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/88913/events",
  "html_url": "https://github.com/kubernetes/kubernetes/issues/88913",
  "id": 577071997,
  "labels": [
    {
      "color": "e11d21",
      "default": false,
      "description": "Categorizes issue or PR as related to a bug.",
      "id": 105146071,
      "name": "kind/bug",
      "node_id": "MDU6TGFiZWwxMDUxNDYwNzE=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/kind/bug"
    },
    {
      "color": "d2b48c",
      "default": false,
      "description": "Categorizes an issue or PR as relevant to SIG Node.",
      "id": 173493665,
      "name": "sig/node",
      "node_id": "MDU6TGFiZWwxNzM0OTM2NjU=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/sig/node"
    },
    {
      "color": "d2b48c",
      "default": false,
      "description": "Categorizes an issue or PR as relevant to SIG Windows.",
      "id": 422106010,
      "name": "sig/windows",
      "node_id": "MDU6TGFiZWw0MjIxMDYwMTA=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/sig/windows"
    }
  ],
  "labels_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/88913/labels{/name}",
  "locked": false,
  "milestone": null,
  "node_id": "MDU6SXNzdWU1NzcwNzE5OTc=",
  "number": 88913,
  "performed_via_github_app": null,
  "repository_url": "https://api.github.com/repos/kubernetes/kubernetes",
  "state": "closed",
  "title": "Pods not starting on Windows nodes",
  "updated_at": "2020-03-16T16:50:01Z",
  "url": "https://api.github.com/repos/kubernetes/kubernetes/issues/88913",
  "user": {
    "avatar_url": "https://avatars2.githubusercontent.com/u/7319830?v=4",
    "events_url": "https://api.github.com/users/adelina-t/events{/privacy}",
    "followers_url": "https://api.github.com/users/adelina-t/followers",
    "following_url": "https://api.github.com/users/adelina-t/following{/other_user}",
    "gists_url": "https://api.github.com/users/adelina-t/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/adelina-t",
    "id": 7319830,
    "login": "adelina-t",
    "node_id": "MDQ6VXNlcjczMTk4MzA=",
    "organizations_url": "https://api.github.com/users/adelina-t/orgs",
    "received_events_url": "https://api.github.com/users/adelina-t/received_events",
    "repos_url": "https://api.github.com/users/adelina-t/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/adelina-t/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/adelina-t/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/adelina-t"
  }
}